crates/redis_interface/src/types.rs

```diff
@@ -5,9 +5,36 @@
 
 use common_utils::errors::CustomResult;
 use error_stack::IntoReport;
+use fred::types::RedisValue as FredRedisValue;
 
 use crate::errors;
 
+pub struct RedisValue {
+    inner: FredRedisValue,
+}
+
+impl std::ops::Deref for RedisValue {
+    type Target = FredRedisValue;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl RedisValue {
+    pub fn new(value: FredRedisValue) -> Self {
+        Self { inner: value }
+    }
+    pub fn into_inner(self) -> FredRedisValue {
+        self.inner
+    }
```

Comment:
Why are we exposing values of `fred type`, don't we have conversions and transformations written for this. Shouldn't those be exposed from here?

Reply:
It's needed here https://github.com/juspay/hyperswitch/pull/1086/files#diff-0867d0a0cd2736dd09355b0280de09e00aa7114048b1b45a860318463f8ee48aR59